src/pages/my_page/user_profile/MyProfile.tsx

```diff
@@ -1,46 +1,177 @@
-import { Box, Divider } from '@mui/material';
+import { useState } from 'react';
+import { useLocation } from 'react-router-dom';
+import { useDispatch, useSelector } from 'react-redux';
 import { Helmet } from 'react-helmet-async';
-import { RootState } from '../../../redux/store';
-import { UserInfo } from './UserInfo';
+import { Box, IconButton, TextField, Tooltip, Typography } from '@mui/material';
+import { LoadingButton } from '@mui/lab';
+import EditIcon from '@mui/icons-material/Edit';
+import { ErrorMessage, Field, FieldProps, Form, Formik, FormikProps } from 'formik';
 import { UserOrcid } from './UserOrcid';
 import { UserRoles } from './UserRoles';
-import { BackgroundDiv } from '../../../components/styled/Wrappers';
 import { ResearchProfilePanel } from './ResearchProfilePanel';
+import { RootState } from '../../../redux/store';
+import { setNotification } from '../../../redux/notificationSlice';
+import { BackgroundDiv, InputContainerBox } from '../../../components/styled/Wrappers';
+import { PageSpinner } from '../../../components/PageSpinner';
+import { CristinPerson, FlatCristinPerson } from '../../../types/user.types';
+import { updateCristinPerson } from '../../../api/userApi';
+import { useFetch } from '../../../utils/hooks/useFetch';
+import { UrlPathTemplate } from '../../../utils/urlPaths';
+import { getValueByKey } from '../../../utils/user-helpers';
+import { isErrorStatus, isSuccessStatus } from '../../../utils/constants';
 
 export const MyProfile = () => {
   const { t } = useTranslation();
+  const dispatch = useDispatch();
+  const location = useLocation();
   // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
   const user = useSelector((store: RootState) => store.user)!; // If user has been empty this route would already be blocked
 
+  const currentCristinId = useSelector((store: RootState) => store.user?.cristinId) ?? '';
+  const isPublicPage = location.pathname === UrlPathTemplate.ResearchProfile;
+
+  const personId = isPublicPage
+    ? new URLSearchParams(location.search).get('id') ?? '' // Page for Research Profile of anyone
+    : currentCristinId; // Page for My Research Profile
+
+  const [person, isLoadingPerson, refetchPerson] = useFetch<CristinPerson>({
+    url: personId,
+    errorMessage: t('feedback.error.get_person'),
+  });
+
+  const firstName = getValueByKey('FirstName', person?.names);
+  const lastName = getValueByKey('LastName', person?.names);
+  const personPreferredFirstName = getValueByKey('PreferredFirstName', person?.names);
+  const personPreferredLastName = getValueByKey('PreferredLastName', person?.names);
+  const [editPreferredNames, setEditPreferredNames] = useState(false);
+
+  type CristinPersonFormData = Pick<FlatCristinPerson, 'preferredFirstName' | 'preferredLastName'>;
+
+  const initialValues: CristinPersonFormData = {
+    preferredFirstName: personPreferredFirstName || firstName,
+    preferredLastName: personPreferredLastName || lastName,
```

Comment:
Tror dette kan gi en bug.. 🤔 
Hvis en bruker ikke har satt noe preferred name, vil det egentlige navnet settes som initial values i Formik. Når man da lagrer så vil disse verdiene også persisteres, så da vil man sette preferred name selv om man ikke prøver å gjøre det
```suggestion
    preferredFirstName: personPreferredFirstName,
    preferredLastName: personPreferredLastName,
```

Reply:
Update: Viser seg at backend håndterer dette, og ikke skriver preffered name om det er likt det faktiske navnet